src/query/plan/variable_start_planner.cpp

```diff
@@ -283,6 +331,28 @@ void VaryQueryPartMatching::iterator::SetCurrentQueryPart() {
   if (merge_it_ != merge_end_) {
     current_query_part_.merge_matching = *merge_it_;
   }
+  DMG_ASSERT(filter_it_ != filter_end_ || filter_begin_ == filter_end_,
+             "Either there are no filter matchings or we can always generate"
+             "a variation");
+
+  auto filter_matchings = *filter_it_;
+  auto iterator_cnt = 0;
+  for (auto &filter : current_query_part_.matching.filters) {
+    auto matchings_size = filter.matchings.size();
+
+    std::vector<FilterMatching> new_matchings;
+    new_matchings.reserve(matchings_size);
+
+    for (auto i = 0; i < matchings_size; i++) {
+      new_matchings.push_back(ToFilterMatching(filter_matchings[iterator_cnt]));
```

Comment:
One more question, why is `iterator_cnt` moving differently from `i`

Reply:
Okay, I think I understand, but maybe naming could be better.

Filter_matchings should be all the matchings inside exists filter across all exists filters, so maybe, all_filter_matchings

And iterator_cnt could be named - all_filter_matchings_idx

Reply:
makes sense